File: src/main/java/org/foo/models/Product.java
package org.foo.models;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.foo.enums.ProductUnit;
import org.hibernate.annotations.Where;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "products")
@Where(clause = "is_deleted=false")
public class Product extends BaseEntity{

    private String name;

    private int quantityInStock;

    private int lowLimitAlert;

    @Enumerated(EnumType.STRING)
    private ProductUnit productUnit;

    @ManyToOne(fetch = FetchType.LAZY)
    private Category category;

}

File: lab1-frontend/src/app/common/product.ts
import {ProductUnit} from "../enums/product-unit";
import {CategoryDto} from "./category-dto";

export class Product {
  constructor(
    public id: number,
    public name: string,
    public quantityInStock: number,
    public lowLimitAlert: number,
    public productUnit: ProductUnit,
    public categoryDto: CategoryDto
  ) {}
}
